src/components/sharygin-cubic-graph.js

```diff
@@ -18,8 +18,16 @@ const getScaledPointsList = pointsDict => {
 };
 
 const getLongLine = (x1, y1, x2, y2) => {
-  const dx = x1 - x2;
-  const dy = y1 - y2;
+  let dx, dy;
+  if (x1 !== x2) {
```

Comment:
```suggestion
  if (x1 !== x2 && y1 !== y2) {
```
The cubic actually intersects some vertical lines so I think we need to check both coordinates here.